pkgs/misc/cups/drivers/brother/mfc9332cdw/default.nix

```diff
@@ -0,0 +1,150 @@
+        $out/usr/bin/brprintconf_${model}
+    '';
+
+    installPhase = ''
+      wrapProgram $out/${relpath_lpd}/psconvertij2 \
+        --prefix PATH : ${stdenv.lib.makeBinPath [ gnused coreutils gawk ]}
+
+      wrapProgram $out/${relpath_lpd}/filter${model} \
+        --prefix PATH : ${stdenv.lib.makeBinPath [ ghostscript a2ps file gnugrep which gnused coreutils ]}
+    '';
+
+    meta = with lib; {
+      description = "Brother ${lib.strings.toUpper model} driver";
+      homepage = "http://www.brother.com";
+      license = licenses.unfree;
+      platforms = [ "x86_64-linux" "i686-linux" ];
+      maintainers = with maintainers; [ felixsinger ];
+    };
+  };
+
+  mfc9332cdwcupswrapper = stdenv.mkDerivation rec {
+    version = "1.1.4-0";
+    name = "${model}cupswrapper-${version}";
+    dlfid = "dlf101642";
+    relpath_wrapper = "${relpath_printer}/cupswrapper";
+
+    src = fetchurl {
+      url = "https://download.brother.com/welcome/${dlfid}/${model}_cupswrapper_GPL_source_${version}.tar.gz";
+      sha256 = "4d9a37d0934b3ba2f67f62ad85c3cc212bf10cf891df7f354f60714cbeb0bd0f";
+    };
+
+    buildInputs = [ mfc9332cdwlpr ];
+    nativeBuildInputs = [ makeWrapper ];
+
+    patchPhase = ''
+      wrapper="cupswrapper/cupswrapper${model}"
+
+      substituteInPlace $wrapper \
+        --replace ${"'"}''${device_model}${"'"} "Printers" \
+        --replace ${"'"}''${printer_model}${"'"} "mfc9332cdw"
+
+      substituteInPlace $wrapper \
+        --replace "/usr/share/ppd" "$out/usr/share/ppd" \
+        --replace "/usr/share/cups/model" "$out/usr/share/cups/model" \
+        --replace "/usr/lib/cups/filter" "$out/lib/cups/filter" \
+        --replace "/var/tmp" "$out/var/tmp" \
+        --replace "/${relpath_printer}/cupswrapper" "$out/${relpath_wrapper}" \
+        --replace "/${relpath_printer}/lpd" "${mfc9332cdwlpr}/${relpath_printer}/lpd" \
+        --replace "/${relpath_printer}/inf" "${mfc9332cdwlpr}/${relpath_printer}/inf"
+
+      substituteInPlace $wrapper \
+        --replace "\`cp " "\`cp -p " \
+        --replace "\`mv " "\`cp -p "
+    '';
+
+    buildPhase = ''
+      make -C brcupsconfig/ all
+    '';
+
+    installPhase = ''
+      # create directory layout
+      mkdir -p $out/lib/cups/filter
+      mkdir -p $out/share/cups/model
+      mkdir -p $out/usr/share/ppd/Brother
+      mkdir -p $out/${relpath_wrapper}
+      # temporary needed for creating the wrapper script
+      # through ${relpath_wrapper}/cupswrapper${model}
+      mkdir -p $out/var/tmp
+
+      # copy files to ${relpath_wrapper}
+      cp brcupsconfig/brcupsconfpt1 $out/${relpath_wrapper}
+      cp cupswrapper/cupswrapper${model} $out/${relpath_wrapper}
+      cp PPD/brother_${model}_printer_en.ppd $out/${relpath_wrapper}
+
+      # configure permissions
+      chmod 755 $out/${relpath_wrapper}/brcupsconfpt1
+      chmod 755 $out/${relpath_wrapper}/cupswrapper${model}
+      chmod 644 $out/${relpath_wrapper}/brother_${model}_printer_en.ppd
```

Comment:
you can use `install -m …` to copy these over while specifying the chmod.
There's also `-D`, which you might be able to use to create the directories in first place.